examples/gio_futures/main.rs

```diff
@@ -5,11 +5,38 @@ use std::str;
 
 use futures::prelude::*;
 
-// Throughout our chained futures, we convert all errors to strings
-// via map_err() and print them at the very end.
-//
-// Open the file for reading, and if that succeeds read the whole file from
-// the resulting input stream.
+fn main() {
+    let c = glib::MainContext::default();
+    let l = glib::MainLoop::new(Some(&c), false);
+
+    c.push_thread_default();
+
+    let file = gio::File::new_for_path("Cargo.toml");
+
+    let l_clone = l.clone();
```

Comment:
Can't this use the clone! macro?

Reply:
I only get it working by using await.
That goes a bit against the idea of this example, since it should display the contrast between using raw Futures and the await syntax.
Maybe you can get it working though :)